input/docs/handbook/data-binding/xamarin-forms.md

```diff
@@ -1,63 +1,73 @@
+ 
+```csharp
+public class TheViewModel : ReactiveObject
 {
-    HostScreen = screen ?? Locator.Current.GetService<IScreen>();
-    var canLogin = (this).WhenAnyValue(
-      x => x.Email,
-      x => x.Password,
-      (em, pa) => 
-      {
-          LoginError = "";
-          return !string.IsNullOrWhiteSpace(em) &&
-          !string.IsNullOrWhiteSpace(pa);
-      });
-
-    this.WhenAnyValue(x => x.Email).Subscribe(n => user.username = n);
-    this.WhenAnyValue(x => x.Password).Subscribe(p => user.password = p);
-
-    Register = ReactiveCommand.CreateFromObservable(() => HostScreen.Router.Navigate.Execute(new RegSelectionViewModel()));
-
-    Login = ReactiveCommand.CreateFromTask<object, LoginResponse>(
-        async _ => 
-        {
-          return await App.userManager.Login(user);
-        }, canLogin);
+    private string theText;
 
-    Login.IsExecuting.ToProperty(this, x => x.IsLoading, out _isLoading);
+    public string TheText
+    {
+        get => theText;
+        set => RaiseAndSetIfChanged(ref theText, value);
+    }
+
+    ReactiveCommand<Unit,Unit> TheTextCommand { get; set; }
+
+    public TheViewModel()
+    {
+        TheTextCommand =
+                ReactiveCommand.CreateFromObservable(ExecuteTextCommand);
+    }
 
-    this.WhenAnyObservable(x => x.Login).Subscribe(response => 
+    private IObservable<Unit> ExecuteTextCommand()
     {
-            Debug.WriteLine(response.access_token);
-            LoginError = response.error_description;
-
-            if(response.access_token != null) 
-            {
-                    Debug.WriteLine("From Blob--> " + x);
-                    App.userManager.setToken(response.access_token);
-                    HostScreen.Router.NavigateAndReset.Execute(new HomeViewModel()).Subscribe();
-            }
-    });
+        TheText = "Hello ReactiveUI";
+        return Observable.Return(Unit.Default);
+    }
 }
+```
 
+```xml
+<rxui:ReactiveContentPage
+  x:Class="local:TheContentPage"
+  x:TypeArguments="vm:TheViewModel">
+  <StackLayout>
+    <Entry x:Name="TheTextBox" />
+    <Label x:Name="TheTextBlock" />
+    <Button x:Name="TheTextButton" />
+  </StackLayout>
+</rxui:ReactiveContentPage>
 ```
 
-In the above view model `canLogin` will listen to changes in the email and password fields and can enable the login command if the fields are not empty. One can even do a regex based email validation. One thing would be to bind this command to a login button inside your view like below
+```csharp
+public partial class TheContentPage : ReactiveContentPage<TheViewModel>
+{
+    public ThePage()
+    {
+        InitializeComponent();
+
+        // Setup the bindings.
+        // Note: We have to use WhenActivated here, since we need to dispose the
+        // bindings on XAML-based platforms, or else the bindings leak memory.
+        this.WhenActivated(disposable =>
+        {
+            this.Bind(ViewModel, x => x.TheText, x => x.TheTextBox.Text)
```

Comment:
This was the important thing to me.  Show all the bindings, and show that there is no need to use XAML bindings with ReactiveUI.